apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/EditLimitPrice.tsx

```diff
@@ -1,34 +1,62 @@
 /* eslint-disable react/jsx-newline */
 import {useSwap} from '@yoroi/swap'
+import BigNumber from 'bignumber.js'
 import React from 'react'
 import {StyleSheet, Text, TextInput, View} from 'react-native'
 
 import {Spacer} from '../../../../../components'
+import {useLanguage} from '../../../../../i18n'
 import {useSelectedWallet} from '../../../../../SelectedWallet'
 import {COLORS} from '../../../../../theme'
 import {useTokenInfo} from '../../../../../yoroi-wallets/hooks'
+import {Quantities} from '../../../../../yoroi-wallets/utils'
 import {useStrings} from '../../../common/strings'
+import {useSwapTouched} from './TouchedContext'
 
 export const EditLimitPrice = () => {
   const strings = useStrings()
+  const {numberLocale} = useLanguage()
 
   const wallet = useSelectedWallet()
 
   const {createOrder} = useSwap()
-  const [inputValue, setInputValue] = React.useState(String(createOrder.selectedPool?.price))
+  // TODO: limitChanged should be defined in useSwap
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
I believe we need a helper that based on the state, calculate everyhing from there, and the opposite too, based on the price, calculate the amount to get and dispatch the actions, I think the state is looking good, we are missing these helpers, we got some for the paired->sell/buy, but none for price change/slippage.